scripts/globals/abilities/activate.lua

```diff
@@ -11,16 +11,26 @@ require("scripts/globals/pets")
 require("scripts/globals/msg")
 -----------------------------------
 
-function onAbilityCheck(player,target,ability)
-    if (player:getPet() ~= nil) then
+function onAbilityCheck(player, target, ability)
+    if player:getPet() ~= nil then
         return tpz.msg.basic.ALREADY_HAS_A_PET,0
-    elseif (not player:canUseMisc(tpz.zoneMisc.PET)) then
+    elseif not player:canUseMisc(tpz.zoneMisc.PET) then
         return tpz.msg.basic.CANT_BE_USED_IN_AREA,0
     else
         return 0,0
     end
 end
 
-function onUseAbility(player,target,ability)
+function onUseAbility(player, target, ability)
+    local effect = tpz.effect.DEBILITATION
+
     player:spawnPet(tpz.pet.id.AUTOMATON)
+
+    if player:hasStatusEffect(effect) then
```

Comment:
it might be worth a helper function to perform this task. do the pets have to be resummoned to update their status in salvage?  know there is a potentially similar situation down the road in voidwatch and also abyssea with buffs getting copied to pets.

Reply:
You start salvage with the debuffs when entering. So you summon after the effect is already in place. And when so the pet will gain what ever effect and power you have. Also using a cell will remove the effect from the player amd the pet at same time.

Reply:
I mean synchronizing pets with masters in general though. it might be worthwhile to just have a reusable something we can call to sync their effects up in these situations rather than copying same block code to each pet script. A global func you feed which effects are being copied to and it do the work all in one spot.

Reply:
I'm probably not describing it well. think mixin like.